### PR TITLE
changed to one browserservermsg

### PR DESCRIPTION
**Pull request by ezhou365**

### CODE REVIEW

**Comment by ezhou365** [comment]
#build

File: dev/com.ibm.ws.security.oauth/src/com/ibm/ws/security/oauth20/web/RegistrationEndpointServices.java [hunk 1]
```diff
@@ -235,28 +225,19 @@ private void processPost(OAuth20Provider provider, HttpServletRequest request, H
...
         if (newClient == null) {
-            String browserErrorMsg = Tr.formatMessage(tc, locales, "OAUTH_REGISTRATION_REQUEST_MISSING_CLIENT");
-            String serverErrorMsg = TraceNLS.getFormattedMessage(RegistrationEndpointServices.class,
-                    MESSAGE_BUNDLE,
-                    "OAUTH_REGISTRATION_REQUEST_MISSING_CLIENT",
-                    null,
-                    "CWWKS1463E: The OpenID Connect registration request does not contain a client. Ensure that the request body is not empty and contains a client encoded in JSON format.");
-            Tr.error(tc, serverErrorMsg);
-            throw new OidcServerException(browserErrorMsg, OIDCConstants.ERROR_INVALID_REQUEST, HttpServletResponse.SC_BAD_REQUEST);
+            BrowserAndServerLogMessage errorMsg = new BrowserAndServerLogMessage(tc, locales, "OAUTH_CLIENT_REGISTRATION_REQUEST_MISSING_CLIENT");
```
Comment (ayoho) [inline dev/com.ibm.ws.security.oauth/src/com/ibm/ws/security/oauth20/web/RegistrationEndpointServices.java]: The message key in the update (`OAUTH_CLIENT_REGISTRATION_REQUEST_MISSING_CLIENT`) doesn't match the original key (`OAUTH_REGISTRATION_REQUEST_MISSING_CLIENT`).

File: dev/com.ibm.ws.security.oauth/src/com/ibm/ws/security/oauth20/web/RegistrationEndpointServices.java [hunk 2]
```diff
@@ -664,14 +609,9 @@ private static String computeRegistrationUri(HttpServletRequest request, String
...
         if (client == null) {
-            String browserErrorMsg = Tr.formatMessage(tc, locales, "OAUTH_CLIENT_REGISTRATION_CLIENTID_NOT_FOUND", new Object[] { clientId });
-            String serverErrorMsg = TraceNLS.getFormattedMessage(RegistrationEndpointServices.class,
-                    MESSAGE_BUNDLE,
-                    "OAUTH_CLIENT_REGISTRATION_CLIENTID_NOT_FOUND",
-                    new Object[] { clientId },
-                    "CWWKS1424E: The client id {0} was not found.");
-            Tr.error(tc, serverErrorMsg);
-            throw new OidcServerException(browserErrorMsg, OIDCConstants.ERROR_INVALID_CLIENT, HttpServletResponse.SC_NOT_FOUND);
+            BrowserAndServerLogMessage errorMsg = new BrowserAndServerLogMessage(tc, locales, "OAUTH_CLIENT_REGISTRATION_PUBLIC_CLIENT_NOT_FOUND", new Object[] { clientId });
```
Comment (ayoho) [inline dev/com.ibm.ws.security.oauth/src/com/ibm/ws/security/oauth20/web/RegistrationEndpointServices.java]: The message key in the update (`OAUTH_CLIENT_REGISTRATION_PUBLIC_CLIENT_NOT_FOUND`) doesn't match the original key (`OAUTH_CLIENT_REGISTRATION_CLIENTID_NOT_FOUND`).

File: dev/com.ibm.ws.security.oauth/src/com/ibm/ws/security/oauth20/web/RegistrationEndpointServices.java [hunk 3]
```diff
@@ -680,14 +620,10 @@ private OidcOAuth20Client validateClientIdExists(String clientId, OidcOAuth20Cli
...
         if (OidcOAuth20Util.isNullEmpty(clientId)) {
-            String browserErrorMsg = Tr.formatMessage(tc, locales, "OAUTH_CLIENT_REGISTRATION_MISSING_CLIENTID", new Object[] { request.getMethod(), OAuth20Constants.CLIENT_ID });
-            String serverErrorMsg = TraceNLS.getFormattedMessage(RegistrationEndpointServices.class,
-                    MESSAGE_BUNDLE,
-                    "OAUTH_CLIENT_REGISTRATION_MISSING_CLIENTID",
-                    new Object[] { request.getMethod(), OAuth20Constants.CLIENT_ID },
-                    "CWWKS1426E: The {0} operation failed as the request did not contain the {1} parameter.");
-            Tr.error(tc, serverErrorMsg);
-            throw new OidcServerException(browserErrorMsg, OIDCConstants.ERROR_INVALID_REQUEST, HttpServletResponse.SC_BAD_REQUEST);
+            BrowserAndServerLogMessage errorMsg = new BrowserAndServerLogMessage(tc, locales, "OAUTH_CLIENT_REGISTRATION_MISSING_CLIENT_ID", new Object[] { clientId });
```
Comment (ayoho) [inline dev/com.ibm.ws.security.oauth/src/com/ibm/ws/security/oauth20/web/RegistrationEndpointServices.java]: The message key in the update (`OAUTH_CLIENT_REGISTRATION_MISSING_CLIENT_ID`) doesn't match the original key (`OAUTH_CLIENT_REGISTRATION_MISSING_CLIENTID`).

**Comment by ezhou365** [comment]
#spawn.fullfat.buckets=com.ibm.ws.security.openidconnect.server_fat.endpoint.clientregistration
#build

**Comment by ezhou365** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.openidconnect.server_fat.endpoint.clientregistration

**Pull request merged**